props/hev.h

```diff
@@ -0,0 +1,480 @@
+  HEVTimer armor_increase_timer_;
+
+  enum Hazard {
+    HAZARD_NONE = 0,
+    HAZARD_BIO = 1,
+    HAZARD_RAD = 2,
+    HAZARD_BLO = 3,
+    HAZARD_CHE = 4,
+    HAZARD_HEA = 5,
+    HAZARD_SHO = 6,
+  };
+
+  Hazard current_hazard_ = HAZARD_NONE;
+
+  // Calculate Physical Damage
+  void DoDamage(int damage, bool quiet = false) {
+    int tens = health_ / 10;
+    if (armor_ >= damage) {
+      armor_ -= ceilf(damage * 0.80 / 2);
+      health_ -= ceilf(damage * 0.20);
+    } else if (armor_ > 0) {
+      int excess_physical = damage - armor_;
+      health_ -= excess_physical;
+      // Make armor compromised sound play before setting armor to 0
+      SaberBase::DoEffect(EFFECT_USER2, 0.0, 0);
+      PVLOG_NORMAL << "Armor Compromised!\n";
+      armor_ = 0;
+    } else {
+      health_ -= damage;
+    }
+
+    // Enforce minimum values
+    if (health_ < 0) health_ = 0;
+    if (armor_ < 0) armor_ = 0;
+
+    // Damage
+    if (!quiet) SaberBase::DoEffect(EFFECT_STUN, 0.0, damage);
+    
+    // Death Sound
+    if (health_ == 0) {
+      SaberBase::DoEffect(EFFECT_EMPTY, 0.0, 100);
+      return;
+    }
+    
+    // Health Alert - only plays when health enters a new multiple of 10
+    int new_tens = health_ / 10;
+    if (tens != new_tens) {
+      SaberBase::DoEffect(EFFECT_USER1, 0.0, 0);
+    }
+
+    // Print Damage, Health and Armor
+    PVLOG_NORMAL << "DAMAGE: -" << damage << "\n";
+    PVLOG_NORMAL << "HEALTH: " << health_ << " / ";
+    PVLOG_NORMAL << "ARMOR: " << armor_ << "\n";
+  }
+
+  // Armor Readout
+  void armor_readout() {
+    PVLOG_NORMAL << "Current Armor: " << armor_ << "\n";
+    
+    // Play random "fuzz" sound only if armor is above 0
+    if (armor_ > 0) {
+      if (random(2) == 0) {
+          hybrid_font.PlayCommon(&SFX_fuzz_high);
+      } else {
+          hybrid_font.PlayCommon(&SFX_fuzz_low);
+      }
+
+      // Small delay to ensure fuzz finishes before armor readout plays.
+      delay(475);
+    }
+
+    // Play armor value
+    SFX_armor.SelectFloat(armor_ / 100.0);
+    hybrid_font.PlayCommon(&SFX_armor);
+  }
+
+  // Clashes
+  void Clash(bool stab, float strength) override {
+    // Don't allow Clashes if dead.
+    if (health_ == 0) {
+      return;
+    }
+
+    // If HEVTimer and PropBase's clash_timeout_ are true to activate Clash.
+    // Otherwise return early.
+    if (clash_timer_.active_ && !clash_timer_.check(this->clash_timeout_)) {
+      return;
+    }
+    
+    // Forward Clash event. No Stabs!
+    PropBase::Clash(false, strength);
+
+    // Damage is based on strength, capped at 50
+    int damage = std::min((int)(strength * 4), 50);
+    float v = (strength - GetCurrentClashThreshold()) / 3;
+
+    // Play Clash sounds based on strength
+    SFX_clash.SelectFloat(v);
+    SFX_clsh.SelectFloat(v);
+    SFX_stab.SelectFloat(v);
+
+    // Play Armor Alarm if Damage is 30 or more
+    if (damage >= 30) {
+      hybrid_font.PlayPolyphonic(&SFX_armor_alarm);
```

Comment:
This is going to play at the same time as whatever DoDamage plays, I suspect that is not what you want?